### PR TITLE
fix issue 19661 - SIGSEGV in std.traits.isFunction

### PR DESCRIPTION
**Pull request by ghost**
The ICE didn't happen when the imported template, isFunction, was copied in the same module as the test case. Now in both cases the same compiler error is issued (and in both cases the line information misses).

### CODE REVIEW

File: src/dmd/initsem.d [hunk 1]
```diff
@@ -374,7 +374,8 @@ extern(C++) Initializer initializerSemantic(Initializer init, Scope* sc, Type t,
             sc = sc.endCTFE();
         if (i.exp.op == TOK.error)
         {
-            return new ErrorInitializer();
+            i.exp = new ErrorExp();
```
Comment (WalterBright) [inline src/dmd/initsem.d]: There's something else wrong here. `.op` should NEVER be `TOK.error` while not being an `ErrorExp`. Need to find out where `.op` was set to `TOK.error` for a proper fix.

**Review by WalterBright — CHANGES_REQUESTED** [review]
Fix is in wrong place.

File: src/dmd/initsem.d (hunk 1 above)
Comment (Basile-z) [reply src/dmd/initsem.d]: Yeah but it IS  an error exp actually. I can return `i` directly after asserting that it is.

File: src/dmd/initsem.d [hunk 2]
```diff
@@ -374,7 +374,8 @@ extern(C++) Initializer initializerSemantic(Initializer init, Scope* sc, Type t,
             sc = sc.endCTFE();
         if (i.exp.op == TOK.error)
         {
-            return new ErrorInitializer();
+            assert(cast(ErrorExp) i.exp);
```
Comment (Geod24) [inline src/dmd/initsem.d]: This `cast` is useless, because `ErrorExp` is an `extern(C++)` class

Comment (WalterBright) [reply src/dmd/initsem.d]: Casting it is even less correct. Again, need to find where i.exp.op was set to TOK.error.

**Review by WalterBright — CHANGES_REQUESTED** [review]

**Comment by RazvanN7** [comment]
New fix : https://github.com/dlang/dmd/pull/9652

**Pull request closed without merging**